bootstrap/kubeadm/api/v1beta1/conversion.go

```diff
@@ -68,6 +68,11 @@ func Convert_v1beta2_KubeadmConfigStatus_To_v1beta1_KubeadmConfigStatus(in *boot
 		out.FailureMessage = in.Deprecated.V1Beta1.FailureMessage
 	}
 
+	// Move initialization to old fields
+	if in.Initialization != nil {
+		out.Ready = in.Initialization.DataSecretCreated
```

Comment:
As discussed, we maybe should use the status replicas field instead to set out.ready correctly.